app/controllers/single_page_subscriptions_controller.rb

```diff
@@ -0,0 +1,63 @@
+class SinglePageSubscriptionsController < ApplicationController
+  include AccountHelper
+  include GovukPersonalisation::ControllerConcern
+
+  before_action do
+    head :not_found unless single_page_notifications_feature_enabled?
+  end
+
+  def show
+    content_item = GdsApi.content_store.content_item(topic).to_h
+    if logged_in?
+      user_id = GdsApi.account_api.get_user(
+        govuk_account_session: @account_session_header,
+      ).to_h["id"]
+
+      begin
+        GdsApi.email_alert_api.find_subscriber_by_govuk_account(
+          govuk_account_id: user_id,
+        )
+      rescue GdsApi::HTTPNotFound
```

Comment:
You don't need this, you can just do

```ruby
GdsApi.email_alert_api.link_subscriber_to_govuk_account(
  govuk_account_session: @account_session_header,
)
```

Reply:
Do you mean losing the entire `begin` block?

Is that because `link_subscriber_to_govuk_account` will return or link?

Reply:
Yes, `find_subscriber_by_govuk_account` is only useful if you want to find the user *without* linking them.  If they're already linked, `link_subscriber_to_govuk_account` just returns them.

Reply:
So, since you don't need to do `find_subscriber_by_govuk_account`, you also don't need to do `get_user`.